### PR TITLE
Clang Formatting

### PR DESCRIPTION
**Pull request by Skptak**
Add in a CI-CD action to clang-format files, as well as check for CRLF line endings and trailing whitespace
Add relevant test cases as well.

### CODE REVIEW

**Comment by kstribrnAmzn** [comment]
Couple of things:
1. I'd love to see smaller example files used. The example files alone for this PR are MASSIVE. In total this PR is adding roughly 60k lines. It's not exactly easy to review.
2. It would be nice to have a comment block at the top of the example files explaining how exactly they were generated and what the problem with the file is. While the names are nice, I can't imagine that you change 50k+ lines for formatting failures

**Comment by kstribrnAmzn** [comment]
Approving because I don't see any stand out problems but I obviously haven't read every single line in depth.

**Review by kstribrnAmzn — DISMISSED** [review]

**Comment by Skptak** [comment]
> Couple of things:
> 
>     1. I'd love to see smaller example files used. The example files alone for this PR are MASSIVE. In total this PR is adding roughly 60k lines. It's not exactly easy to review.

Replaced the files with short ones that are 200 lines long at most.
It was an intentional decision to select files from various FreeRTOS Repos:
1. To show that the formatting actions then also work for files we have.
2. To make my life easier by not needing to generate C or H files that had issues, as I just grabbed files that failed checks when implementing the clang-format action.
Additionally, certain FreeRTOS Files will fail to parse through uncrustify using uncrustify version .72. Where the idea was to grab long ones so that if you're testing the action itself we see that these long files will handle the new formatting changes, if one was changing the .clang-format file for example.
But not going to hold up this PR about this since both you and @paulbartell  have requested shorter files.
 
>     2. It would be nice to have a comment block at the top of the example files explaining how exactly they were generated and what the problem with the file is. While the names are nice, I can't imagine that you change 50k+ lines for formatting failures
Do you still feel this way with the current naming convention?
I have also added a README explaining the purpose of these files.

**Review by rawalexe — APPROVED** [review]

File: clang-formatting/filesWithFormattingErrors/goodFileInHere.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#include <stdlib.h>
+#include <time.h>
+
+typedef struct DateAndTime
+{
+    uint64_t hour;
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Comment (bradleysmith23) [inline clang-formatting/filesWithFormattingErrors/goodFileInHere.c:12]: Should this be changed from DateAndTime? I do not see anything related to the date

File: clang-formatting/filesWithFormattingErrors/testWithFormattingError.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#include <stdint.h>
+
+
+typedef struct DateAndTime
+{
+    uint64_t hour;
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Comment (bradleysmith23) [inline clang-formatting/filesWithFormattingErrors/testWithFormattingError.c:13]: Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

File: clang-formatting/filesWithTrailingWhitespace/testWithTrailingWhitespace.c [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+#include <stdint.h>
+
+
+typedef struct DateAndTime
+{
+    uint64_t hour;
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Comment (bradleysmith23) [inline clang-formatting/filesWithTrailingWhitespace/testWithTrailingWhitespace.c:13]: Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

File: clang-formatting/goodFiles/source/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+#include <stdlib.h>
+#include <time.h>
+
+typedef struct DateAndTime
+{
+    uint64_t hour;
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Comment (bradleysmith23) [inline clang-formatting/goodFiles/source/README.md:12]: Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

File: clang-formatting/goodFiles/source/fileWithErrorSource/errorFileInDirectory.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+#include <stdlib.h>
+#include <time.h>
+
+typedef struct DateAndTime
+{
+    uint64_t hour;
+    uint64_t minutes;
+    uint64_t seconds;
+    uint64_t msec;
+} DateAndTime;
```
Comment (bradleysmith23) [inline clang-formatting/goodFiles/source/fileWithErrorSource/errorFileInDirectory.c:12]: Same comment as [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/82/files#r1316479587)

File: clang-formatting/goodFiles/include/formatErrorTest.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+#if( CLK_SOURCE == CLK_SOURCE_PLL ) && ( ENABLE_PLL == 0 )
+    #error  "PLL has been specified as the CLK_SOURCE but ENABLE_PLL is (0). Please set to (1) in file cgc.h and I am supppppppppppppppperrrrrrrrr long"
+#endif
+
+#if( FCLK_FREQUENCY > 50000000L )
+    #error "FCLK_FREQUENCY Error: Please enter a valid divider value"
+#endif
+
+#if(                        ICLK_FREQUENCY                                      >                            100000000L )
```
Comment (bradleysmith23) [inline clang-formatting/goodFiles/include/formatErrorTest.h:39]: remove extra spaces

File: clang-formatting/goodFiles/include/formatErrorTest.h [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+/* Error checking macros for the clock selction and clock enable defines */
+
+#if( ( CLK_SOURCE != CLK_SOURCE_LOCO ) && ( CLK_SOURCE != CLK_SOURCE_HOCO ) && \
+                                ( CLK_SOURCE != CLK_SOURCE_MAIN ) && ( CLK_SOURCE != CLK_SOURCE_SUB ) &&  \
+     ( CLK_SOURCE != CLK_SOURCE_PLL ) )
+    #error "No CLK_SOURCE specified. Please specify a valid CLK_SOURCE";
+#endif
+
+#if( CLK_SOURCE == CLK_SOURCE_HOCO ) && ( ENABLE_HOCO == 0 )
+    #error "HOCO has been specified as the CLK_SOURCE but           dsafa                                            ENABLE_HOCO is (0). Please set to (1) in file cgc.h"
```
Comment (bradleysmith23) [inline clang-formatting/goodFiles/include/formatErrorTest.h:20]: remove extra spaces

File: clang-formatting/filesWithFormattingErrors/goodFileInHere.c (hunk 1 above)
Comment (Skptak) [reply clang-formatting/filesWithFormattingErrors/goodFileInHere.c:12]: Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

File: clang-formatting/goodFiles/source/README.md (hunk 4 above)
Comment (Skptak) [reply clang-formatting/goodFiles/source/README.md:12]: Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

File: clang-formatting/filesWithTrailingWhitespace/testWithTrailingWhitespace.c (hunk 3 above)
Comment (Skptak) [reply clang-formatting/filesWithTrailingWhitespace/testWithTrailingWhitespace.c:13]: Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

File: clang-formatting/goodFiles/source/fileWithErrorSource/errorFileInDirectory.c (hunk 5 above)
Comment (Skptak) [reply clang-formatting/goodFiles/source/fileWithErrorSource/errorFileInDirectory.c:12]: Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

File: clang-formatting/goodFiles/include/formatErrorTest.h (hunk 6 above)
Comment (Skptak) [reply clang-formatting/goodFiles/include/formatErrorTest.h:39]: Entirely a fake C file that is just used to catch formatting errors, any content in the file itself related to code is irrelevant.

**Review by bradleysmith23 — APPROVED** [review]

**Pull request merged**